### ISSUE TITLE
Switch to installing CUTEst via Meson

### ISSUE THREAD

**Issue by jfowkes** [jfowkes/pycutest#84]
CUTEst and SIFDecode are now installable via Meson. We should switch the CI and documentation over from the old hand-crafted make installer in `.install_cutest.sh` to the simple `meson install` approach which requires no updating and is future-proof.

**Comment by jfowkes** [comment]
See the discussion at https://github.com/ralna/CUTEst/issues/55 for details, the plan for now is to keep the default approach rather than the demuxer as that would be a cache-breaking change for a major release.

**Comment by jfowkes** [comment]
Also SIFDecoder now has a standalone direct Fortran interface as an alternative to the `sifdecode` bash script:
https://github.com/ralna/SIFDecode/pull/24
May also be worth switching to this at some point in the future (don't want to break existing installs for now).

**Comment by amontoison** [comment]
@jfowkes A liitle bit off-topic but I suggest to compile the decoded SIF files with `-03` [here](https://github.com/jfowkes/pycutest/blob/9f1844ac7acd6c01deac9929c00fc64ed7736ff4/pycutest/build_interface.py#L223).
My CI builds in the Julia interface now require much less time (around 30% less) with this modification.

**Comment by jfowkes** [comment]
@amontoison good call! I never thought of that, but of course `-O3` or `-O2` would help a lot with PyCUTEst performance!

**Comment by jfowkes** [comment]
Also probably worth repurposing the CUTEST, SIFDECODE and MASTSIF environment variables so that they can be used to point to non-default meson installs (i.e. not into `/usr/local/`) as some users may desire this flexibility.
